neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -123,7 +147,10 @@ def _precompute_convolution_filter_matrix(grid_in,
     r = torch.sqrt(diffs[0] ** 2 + diffs[1] ** 2)
     phi = torch.arctan2(diffs[1], diffs[0]) + torch.pi
 
-    idx, vals = kernel_handle(r, phi)
+    if isinstance(basis_type, str):
+        # if a basis name is provided, instantiate it with the provided kernel shape
+        basis_type = basis_type_classes[basis_type](kernel_shape)
```

Comment:
What would be the alternative? For the user to pass directly a class, instantiated with the correct kernel shape? In this case, I'd keep only the str option.

Reply:
Done